log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -609,7 +610,13 @@ public Configuration setConfiguration(final Configuration config) {
             final ConcurrentMap<String, String> map = config.getComponent(Configuration.CONTEXT_PROPERTIES);
 
             try { // LOG4J2-719 network access may throw android.os.NetworkOnMainThreadException
-                map.putIfAbsent("hostName", NetUtils.getLocalHostname());
+                // LOG4J2-2808 don't block unless necessary
+                map.computeIfAbsent("hostName", new Function<String, String>() {
```

Comment:
You don't have to be so verbose here, use () -> or ::

Reply:
Fixed. 👍